tests/trace_plugin_test.py

```diff
@@ -0,0 +1,91 @@
+from TestHelper import TestHelper
+from Node import Node
+from WalletMgr import WalletMgr
+from core_symbol import CORE_SYMBOL
+
+class TraceApiPluginTest(unittest.TestCase):
+    sleep_s = 1
+    cluster=Cluster(walletd=True, defproduceraPrvtKey=None)
+    walletMgr=WalletMgr(True)
+    accounts = []
+    cluster.setWalletMgr(walletMgr)
+
+    # kill nodeos and keosd and clean up dir
+    def cleanEnv(self) :
+        self.cluster.killall(allInstances=True)
+        self.cluster.cleanup()
+        self.walletMgr.killall(allInstances=True)
+        self.walletMgr.cleanup()
+
+    # start keosd and nodeos
+    def startEnv(self) :
+        account_names = ["alice", "bob", "charlie"]
+        traceNodeosArgs = " --plugin eosio::trace_api_plugin --trace-no-abis --trace-dir=."
+        self.cluster.launch(totalNodes=1, extraNodeosArgs=traceNodeosArgs)
+        self.walletMgr.launch()
+        testWalletName="testwallet"
+        testWallet=self.walletMgr.create(testWalletName, [self.cluster.eosioAccount, self.cluster.defproduceraAccount])
+        self.cluster.validateAccounts(None)
+        self.accounts=Cluster.createAccountKeys(len(account_names))
+        node = self.cluster.getNode(0)
+        for idx in range(len(account_names)):
+            self.accounts[idx].name =  account_names[idx]
+            self.walletMgr.importKey(self.accounts[idx], testWallet)
+        for account in self.accounts:
+            node.createInitializeAccount(account, self.cluster.eosioAccount, buyRAM=1000000, stakedDeposit=5000000, waitForTransBlock=True, exitOnError=True)
+        time.sleep(self.sleep_s)
+
+    def get_block(self, params: str, node: Node) -> json:
+        time.sleep(self.sleep_s)
+        base_cmd_str = ("curl http://%s:%s/v1/") % (TestHelper.LOCAL_HOST, node.port)
+        cmd_str = base_cmd_str + "trace_api/get_block  -X POST -d " + ("'{\"block_num\":%s}'") % params
+        return Utils.runCmdReturnJson(cmd_str)
+
+    def test_TraceApi(self) :
+        node = self.cluster.getNode(0)
+        for account in self.accounts:
+            self.assertIsNotNone(node.verifyAccount(account))
+
+        expectedAmount = Node.currencyIntToStr(5000000, CORE_SYMBOL)
+        account_balances = []
+        for account in self.accounts:
+            amount = node.getAccountEosBalanceStr(account.name)
+            self.assertEqual(amount, expectedAmount)
+            account_balances.append(amount)
+
+        xferAmount = Node.currencyIntToStr(123456, CORE_SYMBOL)
+        trans = node.transferFunds(self.accounts[0], self.accounts[1], xferAmount, "test transfer a->b")
+        transId = Node.getTransId(trans)
+        blockNum = Node.getTransBlockNum(trans)
+
+        self.assertEqual(node.getAccountEosBalanceStr(self.accounts[0].name), Utils.deduceAmount(expectedAmount, xferAmount))
+        self.assertEqual(node.getAccountEosBalanceStr(self.accounts[1].name), Utils.addAmount(expectedAmount, xferAmount))
+        ret_json = self.get_block(blockNum, node)
```

Comment:
before this, call node.getBlock(blockNum) to retrieve the block and verify that you can find the transaction id in the block, this way if this test case ever fails, we will know whether the problem is in trace_api or before it.